app/src/shared/web3/web3Provider.js

```diff
@@ -15,6 +15,12 @@ import FakeProvider from 'web3-fake-provider'
 declare var ethereum: Web3
 declare var web3: Web3
 
+// Disable automatic reload when network is changed in Metamask,
+// reload is handled in GlobalInfoWatcher component
+if (window.ethereum) {
+    window.ethereum.autoRefreshOnNetworkChange = false
```

Comment:
Why not use this one? I'm fine with it, but curious.

Reply:
The automatic reload is being deprecated in Metamask, they actually recommend listening to `window.ethereum.chainChanged` event in newer Metamask versions to detect the network change. But there is also the older `window.ethereum.chainIdChanged` which apparently was a typo and being deprecated along with removing the web3 injection. And there is `window.ethereum.networkChanged` which does the same thing is somehow different 🤷 We could listen to one (or all) of those events instead of polling but at least this one should work regardless of Metamask's quirks.

According to docs, setting this to false should silence the console warning but there it still is. Anyway, the main reason to disable it here is to make sure that we ourselves are handling the network change, not Metamask.

Reply:
That's exactly the explanation I was looking for, sir. Appreciate it! 🙌